website/blog/2024-06-26-llm-as-judge/index.md

```diff
@@ -0,0 +1,522 @@
+
+Here's a glimpse of how they might visualize their evaluation results:
+
+![Gauge Chart](gauge.png)
+
+This dashboard allows Worldwide WanderAgency's team to quickly assess the performance of their AI systems across different metrics, ensuring they're meeting the high standards required for effective global communication.
+
+To demonstrate the power of using LLMs as judges with MLflow evaluate, let's walk through an example of evaluating a language translation model. We'll create a custom metric called "cultural_sensitivity" to assess how well our translation model preserves cultural nuances. 
+
+For the full code example, please refer to the accompanying notebook. Here's a brief overview of the process:
+
+start by installing all the necessary libraries for this demo to work.
+Since mlflow evaluate offers some default metrics we need to install the libraries to calculate those metrics (toxiticy etc..)
+
+```python
+!pip install mlflow==2.14.1 openai  transformers torch torchvision evaluate datasets openai tiktoken fastapi rouge_score textstat tenacity plotly ipykernel nbformat==5.10.4
+
+!pip install --upgrade nbformat
+```
+
+
+We will be using gpt3.5 and gpt4 during this example for that let's start by setting up an open ai key
+```python
+%env OPENAI_API_KEY=<your-api-key>
+```
+
+Import the necessary libraries
+```python
+import mlflow
+import os
+# Run a quick validation that we have an entry for the OPEN_API_KEY within environment variables
+assert "OPENAI_API_KEY" in os.environ, "OPENAI_API_KEY environment variable must be set"
+
+import openai
+import pandas as pd
+```
+
+When using mlflow.evaluate(), your LLM can be either:
+1.	A `mlflow.pyfunc.PyFuncModel()` . Generally speaking an MLflow model. 
+2.	A `python function` that has strings as inputs and outputs a single string. 
+3.	An `MLflow Deployments` endpoint URI.
+4.	Set `model=None` when you only want to provide the data which has already been scored by the model.
+
+For this case let's use an mlflow model.
+
+Let's start by logging our translation model on mlflow.
+For this tutorial let's use a gpt3.5 with a system prompt
+
+```python
+
+system_prompt = "Translate the following sentences into Spanish"
+basic_translation_model = mlflow.openai.log_model(
+    model="gpt-3.5-turbo",
+    task=openai.chat.completions,
+    artifact_path="model",
+    messages=[
+        {"role": "system", "content": system_prompt},
+        {"role": "user", "content": "{user_input}"},
+    ],
+)
+```
+
+Testing the model to make sure it works
+
+```python
+model = mlflow.pyfunc.load_model(basic_translation_model.model_uri)
+
+model.predict("Hello, how are you?")
+
+# Output = ['¡Hola, ¿cómo estás?']
+```
+
+To use mlflow evaluate we need to prepare some data to provide as input to our LLM.
+In this example we will define some common english expressions which we want translated.
+
+```python
+# 1. Prepare evaluation data
+eval_data = pd.DataFrame(
```

Comment:
Can the outputs be shown from running this micro batch of statements through the translator? It might help elucidate why we're doing this and how idiomatic nuanced literal translations can really miss the mark sometimes.

Reply:
Added the dataframe display for the results of the evaluation and some examples on the introduction on how poor translations can happen